tests/network/provider_migration/libprovider.py

```
@@ -31,18 +40,31 @@ def __exit__(self, exc_type, exc_val, exc_tb):
+            self._check_for_vm_status(vm=vm, status=vim.VirtualMachine.PowerState.poweredOn)
+            self._check_for_ip_obtained(vm=vm)
+        except TimeoutExpiredError:
+            LOGGER.error(f"Timeout while trying to power on VM '{vm_name}' or waiting for IP address assignment")
+            raise
+
+        LOGGER.info(f"Successfully powered on VM '{vm_name}'")
+        return vm
 
     def power_off_vm(self, vm_name: str) -> None:
-        vm = self._get_vm_by_name(name=vm_name)
+        LOGGER.info(f"Powering off VM '{vm_name}'")
```

Ruff (0.14.10)

Logging statement uses f-string

(G004)